memgpt/models/pydantic_models.py

```diff
@@ -91,7 +91,7 @@ class PersonaModel(SQLModel, table=True):
 
 class SourceModel(SQLModel, table=True):
     name: str = Field(..., description="The name of the source.")
-    description: str = Field(None, description="The description of the source.")
+    description: Optional[str] = Field(None, description="The description of the source.")
     user_id: uuid.UUID = Field(..., description="The unique identifier of the user associated with the source.")
     created_at: datetime = Field(default_factory=datetime.now, description="The unix timestamp of when the source was created.")
     id: uuid.UUID = Field(default_factory=uuid.uuid4, description="The unique identifier of the source.", primary_key=True)
```

Comment:
I think it would be better to just add a dictionary metadata field here, since we need that anyway, and then do 
```
source.metadata["num_passages"] = ... 
```
at the time that the REST route lists the sources

Reply:
updated